app/jobs/ingest_version_contents_job.rb

```diff
@@ -0,0 +1,25 @@
+class IngestVersionContentsJob < ApplicationJob
+  queue_as :default
+
+  class GemNotFound < RuntimeError; end
+
+  retry_on GemNotFound, wait: :exponentially_longer, attempts: 3
+  retry_on ActiveRecord::RecordNotFound, wait: :exponentially_longer, attempts: 3
+  # discard_on ActiveJob::DeserializationError
+
+  # TODO: Record somewhere when we just can't get the gem at all for some reason.
+  def perform(version_id)
+    version = Version.indexed.find(version_id)
+    Rails.logger.info "Ingesting gem contents for #{version.full_name}"
+
+    gem = RubygemFs.instance.get("gems/#{version.full_name}.gem")
+    raise GemNotFound, "Gem file not found: #{version.full_name}.gem" unless gem
+    package = GemContent::Package.new StringIO.new(gem)
+    version.manifest.store_spec package.spec.to_ruby
+    version.manifest.store_entries package
+  rescue StandardError => e
```

Comment:
instead of doing this in `perform`, add a `rescue_from(StandardError)` and do the the log / re-raise in there